src/test/java/com/puppycrawl/tools/checkstyle/bdd/ParsedViolation.java

```diff
@@ -0,0 +1,58 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2021 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.bdd;
+
+public final class ParsedViolation {
```

Comment:
lets called `InputViolation` as for `InputConfiguration`

Reply:
From Discord:
> Question: Isn't InputConfiguration a bad file name for executable test code? We name all input files as Input*. It may be mistaken as one by accident.

We need to finalize both names.

Reply:
https://github.com/checkstyle/checkstyle/issues/10225

lets call it `TestInputViolation`

Reply:
done